lib/HierarchyCollection/HierarchyCollectionStory.js

```diff
@@ -0,0 +1,45 @@
+import React, { Fragment } from 'react';
+import { storiesOf } from '@storybook/react';
+import { withKnobs, text, array } from '@storybook/addon-knobs';
+import notes from './README.md';
+import { FolderRow, ItemRow, HierarchyCollection } from '..';
+
+const stories = storiesOf('Components', module);
+stories.addDecorator(withKnobs);
+
+const createKnobCollection = index =>
+  array(`Folders (${index})`, [`Folder ${index}`], ',').map(d => (
+    <Fragment>
+      <FolderRow>{d}</FolderRow>
+
+      {array(`Items (${index})`, [`Item ${index}`], ',').map(i =>
+        i ? <ItemRow>{i}</ItemRow> : null
+      )}
+    </Fragment>
+  ));
+
+stories.add(
+  'HierarchyCollection',
+  () => (
+    <HierarchyCollection className={text('ClassName', '')}>
+      <FolderRow>Root folder</FolderRow>
+      <HierarchyCollection>
+        {createKnobCollection(1)}
+
+        <HierarchyCollection>
+          {createKnobCollection(2.1)}
+          {createKnobCollection(2.2)}
+
+          <HierarchyCollection>
+            {createKnobCollection(3)}
+
+            <HierarchyCollection>{createKnobCollection(4)}</HierarchyCollection>
+          </HierarchyCollection>
+        </HierarchyCollection>
+      </HierarchyCollection>
+    </HierarchyCollection>
+  ),
+  {
+    notes: { markdown: notes }
```

Comment:
Thoughts on moving this and the README.md to a `/lib/HierarchyCollection/docs/`?

Reply:
I like this idea, It's still next to the implementation but separated in some way

Reply:
@AmeeMorris I guess my only challenge would be that we've never created more than 1 doc and ideally we don't want to encourage over documenting at this point.

I think it's a little overkill especially if we're going to get the docs to auto-generate too using the info addon.